### PR TITLE
Generate source package for common test code

### PR DESCRIPTION
**Pull request by maiqbal11**
Removing source dependencies for common testing code and adding them as nuget packages. Generating source package with testing code.

### CODE REVIEW

**Review by fabiocav — COMMENTED** [review]
A little confused as this doesn't seem to be complete... Did you have a chance to validate the changes? The changes here still package the artifacts as a regular NuGet package, the difference here is that now this is generating a symbols package with the source, which aids in debugging.

The goal of the changes are:

- Create a source only package. Referencing the package adds the source file to the target project. We can chat about this if you need more information.
- Remove the project dependency that currently exists from other test projects. Those should reference the NuGet package instead. Currently, those are still referencing the common test infrastructure as a project.

It would be helpful and make things a little simpler if you also had a different solution for the common test infrastructure.

**Comment by maiqbal11** [comment]
1. @fabiocav  The `<IncludeSource>true</IncludeSource>` tag generates a package named `Microsoft.Azure.WebJobs.TestCommon.3.0.2.symbols.nupkg` which includes the source and should also include the `.pdb` file with the appropriate debug symbols (as per the docs here: [IncludeSource](https://docs.microsoft.com/en-us/dotnet/core/tools/csproj#includesource)). I validated these changes before pushing and saw the following structure indicating that the source had been packaged. However, it does look like the debug symbols are currently missing so I'll take a look at that. 

sourcepkg

2. I intentionally didn't remove the project dependencies that exist from other test projects. I was able to verify that solution built correctly with the imported  `TestCommon` nuget package by using a local feed. However, I wanted to use the CI to generate a nuget package from my changes, publish that to MyGet and then perhaps have a separate PR to remove the project dependencies from the other test projects.

3. Could you elaborate more on what you mean by having "a different solution for the common test infrastructure"? Would this purely for local testing or is the intention to have the common test code live in its own solution independent from the WebJobs solution?

**Comment by maiqbal11** [comment]
Reading through your comments a second time, I think I have a better understanding of what the goal is. Somehow, I didn't read much into the "source only" part of your answer. I'll look into how we can get the source into the package with symbols minus the assemblies.

**Comment by fabiocav** [comment]
1 - As I mentioned in my original comment, this is the symbols package, with source included to add source. This is not what we want here. That is not a standalone package that may be referenced. The difference with that package is that you'll have debugging symbols and will be able to step into the source, which we already support with embedded PDBs and source link (the embedded PDBs model is likely why you're not seeing the actual files). We want to generate a **source only** main package, and the proper validation for this is to add a reference with your local build artifacts. What you'll see is that the actual source becomes part of the project.

2 - It's fine to iterate on this, but the goal of our initial set of changes is to have all those projects referencing the NuGet.

3 - Yes. This could be viewed as something that is maintained separately and could conceivably even like in its own repository.

Again, if you don't find a lot of information on source only packages, I can work with you and share some examples/starter project as well.

**Review by brettsam — APPROVED** [review]
This looks good to me... ideally we'd be able to immediately consume these sources in our other repos (once published) and start removing some duplicate helpers.

File: test/Microsoft.Azure.WebJobs.Host.TestHelpers/WebJobs.Host.TestHelpers.csproj [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  <ItemGroup>
+    <PackageReference Remove="@(PackageReference)" />
+  </ItemGroup>
+
+  <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+
+  <Target Name="Compile" />
+  <Target Name="CopyFilesToOutputDirectory" />
+  
+</Project>
```
[the remark this answers is not in the document]
Comment (fabiocav) [reply test/Microsoft.Azure.WebJobs.Host.TestHelpers/WebJobs.Host.TestHelpers.csproj]: Not sure if he used any of the information from there, but we spent a some time walking through this process.

[the remark this answers is not in the document]
Comment (maiqbal11) [reply test/Microsoft.Azure.WebJobs.Host.TestHelpers/WebJobs.Host.TestHelpers.csproj]: I had seen the blog post, but Fabio and I went through the process as well.

**Comment by maiqbal11** [comment]
> Also, I'm still seeing conflicts; be sure to rebase.

I'm not seeing explicit conflicts in the PR itself. Could you elaborate a little more on the kinds of conflicts that you're referring to here? I'll definitely be rebasing when I check in since it's been a while since I last touched this PR. Would there be anything that I need to do beyond that?

**Comment by brettsam** [comment]
It may just be github -- if you squash down your commits, it may go away.

**Comment by maiqbal11** [comment]
> It may just be github -- if you squash down your commits, it may go away.

There was some strange issue where GitHub wasn't showing the rebase conflicts for me but was showing them for you and Fabio. I pulled the latest dev code and did a rebase on my devbox before pushing. Should be resolved now.

**Comment by maiqbal11** [comment]
@fabiocav, did you want to review the PR or should I go ahead and check it in?

**Comment by fabiocav** [comment]
Let me go through this one more time after the latest round of changes. It won't take me long.

**Comment by maiqbal11** [comment]
@fabiocav, could you take a look when you get the chance? Wanted to get this in since it's been open for a while.

**Comment by azfuncgh** [comment]
In an effort to clean up the repository, we're closing stale PRs. If you feel this is still relevant, please reopen the PR and we'll work on moving this forward. Thank you for your contribution!

**Pull request closed without merging**